build-support/bin/_generate_all_lockfiles_helper.py

```diff
@@ -195,6 +212,11 @@ def main() -> None:
         update_default_lockfiles(specified=None)
         return
 
+    if args.pex:
+        update_internal_lockfiles(internal_tools=False)
+        update_default_lockfiles(specified=[Lambdex.options_scope])
```

Comment:
So how does one know this? IOW if we add a tool that has a (transitive maybe even) dependency on Pex or adds one down the line, does that tool need to be added here too?

Reply:
For Lambdex, all we know is this:

https://github.com/pantsbuild/pants/blob/d02cfd6783f6a84934bb53fa075a9bd25b22a49e/src/python/pants/backend/python/subsystems/lambdex.py#L18

Which happens to transitively depend on Pex. So, currently, we can't automate this knowledge.

> does that tool need to be added here too?

Yeah, it should be. But note that we don't _have_ to necessarily bump those default tool lockfiles -- the direct requirements of the tool have stayed the same. They'll still work, only won't be using the newest.

The only thing we actually need to regnenerate is our user lockfile, which is because we bumped `requirements.txt` so now the lockfile is stale vs. our direct requirements: our `python_requirement` says we need pex==2.1.65, but lockfile is still using pex==2.1.64. We have lockfile validation code that will complain if you don't update (altho still a WIP to wire that up cleanly for user lockfiles).

Reply:
Ok. This situation should be improvable once Pex locking lands since it does support -p. IOW, there could be a mode that runs against all lockfiles only adjusting the pex dependency. If there is no (transitive) dependency for a lockfile, it'll stay pinned as-is.